packages/ui/components/listIconText/ListIconText.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { StyleSheet } from "react-native";
+import { tw } from "../../tailwind";
+import { useIsDesktopDevice } from "../../hooks/useIsDesktopDevice/useIsDesktopDevice";
+import { HStack, IStackProps } from "native-base";
+import { Icon, IconNames } from "../icon/Icon";
+import { ListText } from "../listText/ListText";
+
+// TODO either iconName or avatar
```

Comment:
leftover or will you tackle it later?

Reply:
want to tackle it later, but not sure if it is necessary .. thought of doing an xor, what do you think @nikgraf  ?

Reply:
yeah xor sounds good, in Typescript it should work with https://basarat.gitbook.io/typescript/type-system/discriminated-unions

```tsx
export type ListIconTextProps = IStackProps & {
  type: "iconName"
  main: string;
  iconName: IconNames;
  secondary?: string;
} | {
  type: "avatar"
  main: string;
  avatar: React.ReactNode;
  secondary?: string;
};
```